coverage/src/test/kotlin/org/rust/coverage/RsCoverageTest.kt

```diff
@@ -0,0 +1,222 @@
+                }                               // Hits: 1
+            """)
+
+            rust("main.rs", """
+                use hello::foo;
+
+                fn main() {                     // Hits: 1
+                    foo();                      // Hits: 1
+                }                               // Hits: 1
+            """)
+        }
+    }
+
+    fun `test for`() = doTest {
+        toml("Cargo.toml", """
+            [package]
+            name = "hello"
+            version = "0.1.0"
+            authors = []
+        """)
+
+        dir("src") {
+            rust("main.rs", """
+                fn main() {                         // Hits: 1
+                    for i in 1..5 {                 // Hits: 5
+                        println!("Hello, world!");  // Hits: 4
+                    }
+                }                                   // Hits: 1
+            """)
+        }
+    }
+
+    fun `test for function call`() = doTest {
+        toml("Cargo.toml", """
+            [package]
+            name = "hello"
+            version = "0.1.0"
+            authors = []
+        """)
+
+        dir("src") {
+            rust("lib.rs", """
+                fn foo() {                      // Hits: 4
+                    println!("Hello, world!");  // Hits: 4
+                }                               // Hits: 4
+                
+                fn main() {                     // Hits: 1
+                    for _ in 1..5 {             // Hits: 5
+                        foo();                  // Hits: 4
+                    }
+                }                               // Hits: 1
+            """)
+        }
+    }
+
+    fun `test tests`() = doTest(runTests = true) {
+        toml("Cargo.toml", """
+            [package]
+            name = "hello"
+            version = "0.1.0"
+            authors = []
+        """)
+
+        dir("src") {
+            rust("lib.rs", """
+                fn foo() {                      // Hits: 2
+                    println!("Hello, world!");  // Hits: 2
+                }                               // Hits: 2
+                
+                #[test]
+                fn test1() {                    // Hits: 2
+                    foo();                      // Hits: 1
+                }                               // Hits: 2
+                
+                #[test]
+                fn test2() {                    // Hits: 2
+                    foo();                      // Hits: 1
+                }                               // Hits: 2
+            """)
+        }
+    }
+
+    private fun executeGrcov(configuration: RunConfiguration) {
+        val executor = DefaultRunExecutor.getRunExecutorInstance()
+        val environment = ExecutionEnvironmentBuilder.create(executor, configuration).build()
+        val state = environment.state as CargoRunStateBase
+        GrcovRunner().execute(environment, null, state)
```

Comment:
Hm, looks like environment already should contain correct runner (of course, if you use correct executor)